### DISCUSSION TITLE
Missing Ability to reset pybind11 ?

### DISCUSSION THREAD

**Discussion by gsohler** [pybind/pybind11#4947] [General]
Hi,
In my project i use embedded python and this uses a python wrapper which is made with pybind11.
The Application needs to start the embedded python interpreter several times and also finishes it nicely with Py_FinalizeEx()

Now the problem is:
* embedded python is started the 2nd time, 
*  python starts
*  python imports a module made with pybind11
*  Now I get the error: ImportError: generic_type: type "BRepLib_ShellError" is already registered!

I believe that the problem is in

inline internals **&get_internals_pp(){
     static internals **internals_pp = nullptr;
}

I could not find any mechanism, which can reset this structure. This file should be reset at the same time when i call Py_FinalzeEx()

if that exist, please let me know!

*

**Comment by evguran** [comment]
As far as I get you use pybind11 to embed python into your application. If this is the case you should use `py::finalize_interpreter` to properly close the interpreter
